roles/nova-common/meta/main.yml

```diff
@@ -3,11 +3,13 @@ dependencies:
   - role: endpoints
   - role: apt-repos
     repos:
-      - repo: 'deb {{ apt_repos.cloud_archive.repo }} precise-updates/icehouse main'
+      - repo: 'deb {{ apt_repos.cloud_archive.repo }} {{ ansible_distribution_release }}-updates/kilo main'
         key_package: ubuntu-cloud-keyring
     when: ansible_distribution_version == "12.04"
```

Comment:
This task is conditional to Precise base OS. Cloud Archive does not support Kilo for Precise, they only went up to icehouse. Making this task templated as you have will break on a Precise system. Perhaps it would be best to remove this task and drop support for Precise with our 2.0 release of Ursula.